trustwallet/wallet-core
# Cardano signing

## Description

Add Cardano transaction signing and protobuf definitions.
Closes #745 .

## Testing instructions

- Create new Cardano address (rec. phrase, derived prikey, addr)
- Send some ADA to it
- check address&txs on https://cardanoexplorer.com
- put together new transaction, sign it with Signer -- see Cardano/SignerTests.cpp.
- push result to node, like this:

    curl -d '{"txHash":"ebf58670ee1512d597876fe632e92790bd70568374bdbe5a69c5d8ed107607ee","txBody":"82839f8200d8185824825820a49bad3f69bbab0e4d3e51991ce7a1116c0fd322a7731246b92df455e67e686100ff9f8282d818582183581c0f1fdc075b3e562ff4c8dc6979b136aeedd087a7d3bf554f774b58eba0001abb7c9a121a007a12008282d818582183581c20b55a0bd5cbc39b58cdd8977e863aca7d387a5a0fc37a473eb72042a0001aa6fa2f8a1a001bece4ffa0818200d8185885825840835610e371e632d829eb63bf44ec39d0487c19d4d98b8dce9dd88d88414ce5bb1ae60957699bf72b212ca823520cf7d86af5d1304cd90248fe60bd1fe442870f584045b3fe46b52c839566e697cbe72e248cd0f01e10c7c5696c8f202fe9776602b63c5878f0e27869d1d9ff69c9773211406970742f7cc316690b9d9133794e4f0e"}' -H "Content-Type: application/json" https://adalite.io/api/txs/submit

Also, see Cardano unit tests.

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [x] Add C-interface impl and tests
- [x] Add swift and kotlin tests
- [x] Handle error cases
- [x] Fill SigningOutput.transaction
- [x] Change to fee-per-byte model
- [x] Leave out change output if 0
- [x] Handle TODOs from code
- [x] Adapt TransactionPlan (like in Bitcoin)
